src/ui/wxWidgets/TreeCtrl.cpp

```diff
@@ -1012,12 +1013,23 @@ void TreeCtrl::OnRenameGroup(wxCommandEvent& WXUNUSED(evt))
   }
 }
 
+void TreeCtrl::OnStartLabelEdit( wxTreeEvent& evt )
+{
+  wxTreeItemId item = evt.GetItem();
+  if(IsReadOnly() || !item.IsOk() ||! ItemIsGroup(item)) {
+    // In case of read only, item not ok or item is no group editing in the tree looks not promissing
+    evt.Veto();
+    return;
+  }
+}
+
 void TreeCtrl::OnEndLabelEdit( wxTreeEvent& evt )
 {
   const wxString &label = (evt.IsEditCancelled() ? GetItemText(evt.GetItem()) : evt.GetLabel());
 
-  if (label.empty()) {
-    // empty entry or group names are a non-no...
+  if (label.empty() || IsReadOnly()) { // end editing without message to avoid error "Dots are not allowed in group names" with Read Only data base and a dot in a shown name
```

Comment:
The check for IsReadOnly() is unnecessary and confusing, since this is checked in OnStartLabelEdit().

Reply:
Ok, removed in last check in to handle in TreeCtrl::OnStartLabelEdit() only.